.github/workflows/publish-npm.yml

```diff
@@ -0,0 +1,32 @@
+
+name: Publish NPM Package
+
+on:
+  push:
+    tags:
+      -  v*
+jobs:
+  build:
+    runs-on: ubuntu-24.04
+    permissions:
+      id-token: write
+      contents: write # required to get token for "npm provenance" and to dispatch event
+    steps:
+      - name: Set VERSION
+        shell: bash
+        run: echo "VERSION=${GITHUB_REF#refs/tags/v*}" >> $GITHUB_ENV
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version-file: '.nvmrc'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Build
+        run: npm run build
+
+      - run: 📦 npm publish --provenance --access public
```

Comment:
I guess the emoji should be used in the name of the step (to be added if needed) and removed from the run directive
Otherwise it will be used in the bash command and it will fail